src/report_reader.cpp

```diff
@@ -82,16 +82,17 @@ void filterTimestampSorted(Spikes& spikes, double tstart, double tstop) {
 }
 
 template <typename T>
-T make_key(NodeID node_id, ElementID element_id);
+void emplace_ids(T& key, NodeID node_id, ElementID element_id);
```

Comment:
We used to have something like `result.ids.emplace_back(make_key<T>(...))` here, because the vector in the result was not resized but "reserved". Now the code needs random access to the output array due to the IO-blocks and we can no longer grow the vector element by element. Thus, we resize the vector first and then copy the values to the actual position according to the positions indexed and stored internally.

I'm not sure how to solve this differently, because `T` can either be a `uint64_t` or a `std::array<uint64_t, 2>`. What would be your approach on this? @matz-e suggested `constexpr`, but it is only available in C++17 and above.

Reply:
I'd just drop the templeting:
```
void emplace_ids(NodeID& key, NodeID node_id, ElementID /* element_id */) {
    key = node_id;
}

void emplace_ids(CompartmentID& key, NodeID node_id, ElementID element_id) {
    key[0] = node_id;
    key[1] = element_id;
}
```

Reply:
Thank you so much, that actually makes sense. I have also added `inline` to your suggestion and verified that the performance is the same (or slightly better even):

```
   GIDs |   Elapsed |      Open |    Select |       Get
      1 |  1.673048 |  1.051501 |  0.619802 |  0.001734
     10 |  1.376986 |  0.721323 |  0.628078 |  0.027575
    100 |  4.257407 |  0.710185 |  0.638011 |  2.909204
   1000 |  7.380188 |  0.777585 |  0.625556 |  5.977037
  10000 |  7.389488 |  0.770053 |  0.642935 |  5.976491
 100000 |  7.823633 |  0.766917 |  0.633076 |  6.423629
1000000 | 11.833009 |  0.752722 |  0.698966 | 10.381311
4234929 | 20.914003 |  0.757421 |  0.268733 | 19.887840
```